hashes/src/lib.rs

```diff
@@ -314,6 +314,15 @@ mod sealed {
     impl<const N: usize> IsByteArray for [u8; N] { }
 }
 
+trait RenameMe: HashEngine {
```

Comment:
Why do we need a new trait for it?

Reply:
It's a private trait that gets us macro-like codegen without needing to use macros.

Reply:
Given that there's only one method though, I think a freestanding function would be even nicer (though a tiny bit less nice to use).

Reply:
Yes, I'd prefer a free-standing function or a private method if the genericity of the trait is not needed to avoid confusion.